src/client/mod.rs

```diff
@@ -318,9 +319,35 @@ where C: Connect + Sync + 'static,
             })
         };
 
-        let race = checkout.select(connect)
-            .map(|(pooled, _work)| pooled)
-            .or_else(|(e, other)| {
+        let executor = self.executor.clone();
+        let race = checkout.select2(connect)
```

Comment:
I would leave a comment here that `.started()` being false depends on the order of the futures in the Select2 struct, given that future `A` is polled before future `B`. So we shouldn't change it to be
`connect.select2(checkout)` as it would break things.

Also if Select2's logic ever changes it would break things, but ¯\\_(ツ)_/¯.

Reply:
Thanks, added a comment for Future Someone.

Yes, this does depend on select's current logic... though the new test would be failed, and there is already another test to make sure the connect future is lazy, since we were otherwise seeing a lot of TCP streams started up and tossed immediately with a custom connector...